rise/YogaPoseAPIManager.m

```diff
@@ -19,28 +20,34 @@ - (id)init {
 }
 
 - (void)fetchPoses:(void(^)(NSArray *poses, NSError *error))completion {
-    NSURL *url = [NSURL URLWithString:@"https://lightning-yoga-api.herokuapp.com/yoga_poses"];
-    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReturnCacheDataElseLoad timeoutInterval:10.0];
-    NSURLSessionDataTask *task = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-        if (error != nil) {
-            NSLog(@"%@", [error localizedDescription]);
-            // The network request has completed, but failed.
-            // Invoke the completion block with an error.
-            completion(nil, error);
-        }
-        else {
-            NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
-
-            NSArray *dictionaries = dataDictionary[@"items"];
-            NSArray *poses = [YogaPose posesWithDictionaries:dictionaries];
-
-            // The network request has completed, and succeeded.
-            // Invoke the completion block with the movies array.
-            // Think of invoking a block like calling a function with parameters
-            completion(poses, nil);
-        }
-    }];
-    [task resume];
+    //look for poses in realm
+    RLMResults<YogaPose *> *poses = [YogaPose allObjects];
+    if (poses.count == 0) {
+        NSURL *url = [NSURL URLWithString:@"https://lightning-yoga-api.herokuapp.com/yoga_poses"];
+        NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReturnCacheDataElseLoad timeoutInterval:10.0];
+        NSURLSessionDataTask *task = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+            if (error != nil) {
+                NSLog(@"%@", [error localizedDescription]);
+                // The network request has completed, but failed.
+                // Invoke the completion block with an error.
+                completion(nil, error);
+            }
+            else {
+                NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+
+                NSArray *dictionaries = dataDictionary[@"items"];
```

Comment:
Every data model downloaded from network should be verified. Firstly check if dataDictionary[@"items"] is nil, secondly check if it's NSArray.